### ISSUE TITLE
Upgrade rusqlite

### ISSUE THREAD

**Issue by theduke** [bikeshedder/deadpool#162]
Rusqlite `0.26` was released a month ago.

Would be great to see a new release for deadpool-sqlite.

**Comment by bikeshedder** [comment]
I tried that two weeks ago (bd4cc5b2c727ed2cbf4cfef336cb250e0110aba5) just to realize that diesel uses an incompatible versino of `libsqlite3-sys`. Therefore I had to revert this change (a095b44ffb9fa0ae983d1a846ac7358a4f5bf8f2).

I just created a branch and PR with that change: #163

```
$ cargo test --workspace --all-features
error: multiple packages link to native library `sqlite3`, but a native library can be linked only once

package `libsqlite3-sys v0.9.1`
    ... which satisfies dependency `libsqlite3-sys = "=0.9.1"` of package `diesel v1.4.8`
    ... which satisfies dependency `diesel = "=1.4.8"` of package `deadpool-diesel v0.3.0 (/home/bikeshedder/projects/foss/deadpool/diesel)`
links to native library `sqlite3`

package `libsqlite3-sys v0.23.1`
    ... which satisfies dependency `libsqlite3-sys = "=0.23.1"` of package `rusqlite v0.26.1`
    ... which satisfies dependency `rusqlite = "=0.26.1"` of package `deadpool-sqlite v0.4.0-pre (/home/bikeshedder/projects/foss/deadpool/sqlite)`
also links to native library `sqlite3`
```

There are currently two solutions to this:

1. Change the CI tests so it doesn't use the `--workspace` flag anymore. While this fixes the CI error it will still cause people to run into the exact same error if they every try to mix `deadpool-diesel` and `deadpool-sqlite`. I don't think this is very likely so I'm open to this option.
2. Update the dependency of `libsqlite3-sys` of `diesel` to the latest version so the conflict is resolved.

**Comment by bikeshedder** [comment]
The PR diesel-rs/diesel#2923 was merged a month ago. So whenever a new `diesel` version is released this version mismatch will go away. I'm marking this a blocked. If anyone needs the new `rusqlite` version you can just use the branch of the PR #163.

**Comment by ohmyarch** [comment]
https://github.com/rusqlite/rusqlite/releases/tag/v0.27.0

**Comment by bikeshedder** [comment]
It's quite unfortunate. `rusqlite` and `diesel` depend on different versions of `libsqlite3-sys` which both depend on different versions of the native library `sqlite3`:

```
❯ cargo test --workspace
error: multiple packages link to native library `sqlite3`, but a native library can be linked only once

package `libsqlite3-sys v0.9.1`
    ... which satisfies dependency `libsqlite3-sys = ">=0.8.0, <0.23.0"` (locked to 0.9.1) of package `diesel v1.4.8`
    ... which satisfies dependency `diesel = "^1.4.7"` (locked to 1.4.8) of package `deadpool-diesel v0.3.1 (/home/bikeshedder/projects/foss/deadpool/diesel)`
links to native library `sqlite3`

package `libsqlite3-sys v0.24.1`
    ... which satisfies dependency `libsqlite3-sys = "^0.24.0"` (locked to 0.24.1) of package `rusqlite v0.27.0`
    ... which satisfies dependency `rusqlite = "^0.27"` (locked to 0.27.0) of package `deadpool-sqlite v0.4.0-pre (/home/bikeshedder/projects/foss/deadpool/sqlite)`
also links to native library `sqlite3`
```

The easiest fix would be to mark `deadpool-diesel` and `deadpool-sqlite` as incompatible. It's quite annoying tbh.

**Comment by ohmyarch** [comment]
Don't care about diesel at all, never use it. Not being able to use the latest rusqlite because of it is unreasonable.

**Comment by bikeshedder** [comment]
> (...) Not being able to use the latest rusqlite because of it is unreasonable.

You are 100% right. I'm just hesitant to upgrade rusqlite as long as it breaks the CI pipeline. I might create a 0.26 branch which just won't get merged into main until this issue is solved.

Anyone wants to jump in and fix the CI so it no longer uses the `--workspace` flag when running cargo commands?

**Comment by bikeshedder** [comment]
I just released `deadpool-sqlite 0.4.0` on crates.io: https://crates.io/crates/deadpool-sqlite/0.4.0

This is related to PR which currently breaks the CI: https://github.com/bikeshedder/deadpool/pull/189

**Comment by Sparkenstein** [comment]
Hello, [0.29.0](https://github.com/rusqlite/rusqlite/releases/tag/v0.29.0) was released recently. Any plans to update?
